### PR TITLE
Hardlight bow traitor item

### PR DESCRIPTION
**Pull request by Aviu00**
## About the PR

Added hardlight bow which has infinite shots and is able to switch between firing modes by pressing Alt+Z. Costs ~60~ 80 tc. There are 7 total modes:
Regular: ~40~ 30 heat damage.
Disabler: 60 stamina damage.
Fiery: ~25~ 15 heat and ~1~ 1.5 fire stack.
Freezing: ~25~ 20 cold and lowers temp by ~30~ 25.
Explosive: No direct damage but explodes.
X-ray: ~30~ 24 heat damage, pierces through anything (no rad damage cause too op).
Bee: Summons 4 weakened bees (2 times as weak compared to normal and deal no poison damage), bees disappear after ~20~ 5 seconds.

## Why / Balance

More variety for traitors.

## Media

https://github.com/user-attachments/assets/8cd1591f-b38e-435d-9c32-0831034bfafd

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl: Aviu
- add: New traitor item hardlight bow.
- tweak: Bows are now automatically unwielded after firing.

## Release Notes

- **New Features**
  - Introduced a **Hardlight Bow** with multiple firing modes and the ability to fabricate hardlight arrows.
  - Added a **radial menu** for ammunition selection, enhancing user experience in weapon management.
  
- **Improvements**
  - Enhanced projectile mechanics, allowing for new behaviors such as ignoring transparent entities and spawning additional entities on hit.
  - Implemented new **selectable ammunition types** with unique effects, including energy, fiery, and explosive arrows.

- **Localization**
  - Added descriptions and names for the Hardlight Bow in the uplink catalog and weapon localization files.

### CODE REVIEW

**Comment by lanse12** [comment]
1 fire stack is like 5 damage btw

**Comment by BombasterDS** [comment]
Also for 60 TC DPS is too high especially with infinite ammo. Make it 80 TC.
Regular: Damage should be 30 or less. 40 is too high.
Disabler: Really, not sure about it. It'll stamina crit anyone with two shoots and first one will slow target down. Better decrease it to 35.
Fiery: Decrease heat damage to 20.
Freezing: Cold damage is too op. Ingame armor don't have protection against it and slow is too high. Decrease damage to 15 or less and temperature to -20.
Explosive: Decrease explosion radius to 1 tile (crest shape). 
Xray: Make projectile disappear after 3-4 seconds, it shouldn't pierce through all the station. Also decrease damage to 20.
Bee: Fine, i guess? Didn't use beenades in game. (I think it may cause a lot of lags after 10 shots with bow, so it's better do decrease bee count to 1-2)

**Comment by lanse12** [comment]
> Also for 60 TC DPS is too high especially with infinite ammo. Make it 80 TC. Regular: Damage should be 30 or less. 40 is too high. Disabler: Really, not sure about it. It'll stamina crit anyone with two shoots and first one will slow target down. Better decrease it to 35. Fiery: Decrease heat damage to 20. Freezing: Cold damage is too op. Ingame armor don't have protection against it and slow is too high. Decrease damage to 15 or less and temperature to -20. Explosive: Decrease explosion radius to 1 tile (crest shape). Xray: Make projectile disappear after 3-4 seconds, it shouldn't pierce through all the station. Also decrease damage to 20. Bee: Fine, i guess? Didn't use beenades in game. (I think it may cause a lot of lags after 10 shots with bow, so it's better do decrease bee count to 1-2)

just make it 80tc and don't change shit this is funny space game

**Comment by Aviu00** [comment]
> Also for 60 TC DPS is too high especially with infinite ammo. Make it 80 TC. Regular: Damage should be 30 or less. 40 is too high. Disabler: Really, not sure about it. It'll stamina crit anyone with two shoots and first one will slow target down. Better decrease it to 35. Fiery: Decrease heat damage to 20. Freezing: Cold damage is too op. Ingame armor don't have protection against it and slow is too high. Decrease damage to 15 or less and temperature to -20. Explosive: Decrease explosion radius to 1 tile (crest shape). Xray: Make projectile disappear after 3-4 seconds, it shouldn't pierce through all the station. Also decrease damage to 20. Bee: Fine, i guess? Didn't use beenades in game. (I think it may cause a lot of lags after 10 shots with bow, so it's better do decrease bee count to 1-2)

This would suck. I'd rather buff sec later than nerf it so much.

**Comment by a-person5660** [comment]
up the fire stacks on the fire mode and make this 80 TC.

**Comment by a-person5660** [comment]
also make the bees do more damage but last 5 seconds to decrease lag.

**Comment by Aviu00** [comment]
Made it 80 TC. Will maybe change something else but not today.

**Comment by JoeHammad1844** [comment]
Can you alt click or use the context menu to change modes?

**Comment by Aviu00** [comment]
> Can you alt click or use the context menu to change modes?

Yes, it opens radial menu.

**Comment by whateverusername0** [comment]
OP, make it 100 tc :godo:

**Comment by Aviu00** [comment]
> OP, make it 100 tc :godo:

So you can't even buy it with noslips... No, I'll nerf it later properly.

**Comment by Aviu00** [comment]
Alright, made some changes.

Regular arrow damage 40 -> 30.
Disabler arrow unchanged. It is just disabler but infinite. In fact, it is actually worse than disabler because it less convenient to use.
Fiery damage 30 -> 15, Fire stacks 1 -> 1.5.
Freezing damage 25 -> 20, now cools down by 25.
Explosion damage weakened slightly
X-ray now despawns after 2 seconds and damage changed 30 -> 24.
Bees are now slightly more powerful but disappear aftter 5 seconds.

Also made it 60 tc again because it now sucks, and also it is now possible to purchase it with noslips and web vest.
Bow has slightly less DPS compared to esword, but it is ranged and costs 20 tc more, I think it's balanced.

**Comment by Aviu00** [comment]
Ok, I just realized that bees are too op. Made them attack everything including syndies and deal no poison damage. But their total damage increased to 1.5 pierce.

**Comment by Piras314** [comment]
I absolutely love this, but it should absolutely be 80-100tc, it's very very good.

**Review by Piras314 — APPROVED** [review]

**Comment by Aviu00** [comment]
> Forgot about the TC thing. I think 80 is reasonable for it. On everything but the TC value though, this is approved

Done.

**Comment by Aviu00** [comment]
Rabbit finally woke up.

File: Content.Shared/_Goobstation/Weapons/AmmoSelector/SelectableAmmoSystem.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if (!_protoManager.TryIndex(proto, out var index))
+            return false;
+
+        ent.Comp.CurrentProto = proto;
+        provider.Proto = index.ProtoId;
+
+        if (index.Color != null && TryComp(ent, out AppearanceComponent? appearance))
+            _appearance.SetData(ent, ToggleableLightVisuals.Color, index.Color, appearance);
+
+        Dirty(ent);
```
[the remark this answers is not in the document]
Comment (Aviu00) [reply Content.Shared/_Goobstation/Weapons/AmmoSelector/SelectableAmmoSystem.cs:62]: This is correct usage. `Dirty` method has override for `Entity<Component>`

File: Resources/Prototypes/_Goobstation/Entities/Objects/Weapons/Guns/Bow/hardlight_arrows.yml [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  - type: PointLight
+    color: yellow
+  - type: Projectile
+    deleteOnCollide: true
+    damage:
+      types:
+        Heat: 0
+  - type: SpawnEntitiesOnHit
+    proto: MobAngryBeeWeak
+    amount: 4
```
[the remark this answers is not in the document]
Comment (Aviu00) [reply Resources/Prototypes/_Goobstation/Entities/Objects/Weapons/Guns/Bow/hardlight_arrows.yml:190]: `TimedDespawn` handles that.

**Comment by Aviu00** [comment]
Just realized that `ChangeTemperatureOnCollide` component existed so I didn't need to write another one.
Now freezing arrow cools down by slightly less than 25 for humans and slightly more for species with lower mass.

**Comment by Aviu00** [comment]
Ready to merge.

**Pull request merged**